server/server/src/file_service.rs

```diff
@@ -216,8 +216,9 @@ pub async fn change_doc(
             return Err(ClientError(DocumentDeleted));
         }
 
+        tx.last_seen.insert(owner, get_time().0 as u64);
```

Comment:
I think you additionally want to do this in `upsert` if they're re-organizing their stuff, they're active

Reply:
You don't want to flag them as active here, you want to flag them as active in the second transaction. 

Observe the details of this function, it's modeling what's known as the 2 phased commit.

there's files on disk that we're trying to keep in sync with the contents of our database.

And this function can be interrupted at any point and still be fine:

1. we do some prelim checks to make sure they can update this file
2. we write the new file
3. we update our database to point to a new file
4. we cleanup the old file

if we were naive about it we would just do this:

1. overwrite the old file
2. update the pointer in our db

or 

1. update the pointer
2. overwrite the file

if either of those operations fail, we point to a file that doesn't exist 

in our 2 phased commit strategy, we can recover from a failure at any stage and the worst outcome is that we have a file that we need to cleanup.